### PR TITLE
Fix #5773 - FromSql in subqueries

### LINKED ISSUE #5773: FromSql in subqueries
Support for `FromSql` in subqueries.

**Example 1**

```c#
from p in context.Posts
where context.Blogs.FromSql("SELECT * FROM Blogs").Select(b => b.Id).Contains(p.BlogId)
select p
```

Currently, this query evaluates the where clause client-side, causing repeated execution of the `FromSql` query.

**Example 2**

```c#
from p in context.Posts
where p.IsPublished &&
context.Blogs.FromSql("SELECT * FROM Blogs").Select(b => b.Id).Contains(p.BlogId)
select p
```

Currently, this query throws `NotImplementedException` from `ResultOperatorHandler.HandleResultOperator`.

1.0.0-rc2-final, Windows 10, VS 2015

### PR DESCRIPTION
**Pull request by anpete**
- Separated requires client eval from composability for FromSql queries in order that FromSql queries can be lifted.
- Fixed a broken assumption in QueryAnnotationExtractor: Extraction should do a deep search of query model expressions.

### CODE REVIEW

**Comment by anpete** [comment]
@maumar @smitpatel

File: src/Microsoft.EntityFrameworkCore.Relational/Query/ExpressionVisitors/RelationalEntityQueryableExpressionVisitor.cs [hunk 1]
```diff
@@ -196,6 +196,8 @@ var useQueryComposition
                         .TrimStart()
                         .StartsWith("SELECT ", StringComparison.OrdinalIgnoreCase);
 
+                var requiresClientEval = !useQueryComposition;
```
Comment (maumar) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/ExpressionVisitors/RelationalEntityQueryableExpressionVisitor.cs]: nit: move this a bit lower

File: src/Microsoft.EntityFrameworkCore.Relational/Query/Expressions/SelectExpression.cs [hunk 2]
```diff
@@ -133,6 +133,19 @@ var selectExpression
         public virtual bool IsProjectStar { get; set; }
 
         /// <summary>
+        /// </summary>
```
Comment (maumar) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/Expressions/SelectExpression.cs]: missing docs

**Comment by maumar** [comment]
:shipit:

File: src/Microsoft.EntityFrameworkCore.Relational/Query/Expressions/SelectExpression.cs (hunk 2 above)
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/Expressions/SelectExpression.cs]: Fixing.

**Pull request merged**